src/ElectricPokemon.java

```diff
@@ -0,0 +1,46 @@
+public class ElectricPokemon extends Pokemon {
+    private final String type;
```

Comment:
Je hebt in elke klasse een Private final String type; neergezet. Misschien is hier een makkelijkere oplossing voor ;)